tempesta_fw/ss_skb.c

```diff
@@ -498,16 +550,12 @@ __split_pgfrag_del(struct sk_buff *skb, struct sk_buff *pskb, int i, int off,
 	}
 
 	return skb_frag_address(&skb_shinfo(skb_dst)->frags[i]);
+
 lookup_next_ptr:
-	/* Try to find next data chunk after deleted fragment. */
+	/* Get the next fragment after the deleted fragment. */
 	if (i < si->nr_frags)
 		return skb_frag_address(&si->frags[i]);
-	skb_dst = skb_shinfo(pskb ? : skb)->frag_list;
-	if (skb_headlen(skb_dst))
-		return skb_dst->data;
-	if (skb_shinfo(skb_dst)->nr_frags)
-		return skb_frag_address(&skb_shinfo(skb_dst)->frags[0]);
-	/* ...not found, return invalid address try next skb. */
```

Comment:
We get here when we don't have knowledge of where the next data is. Thus this `return (void *)1`. How do you propose we can eliminate that?